<!--
-->
<launch>
  
  <!-- Use simulation time for TFs -->
  <param name="/use_sim_time" value="true" /> 

  <group ns="dll_ns">
    <!-- Fake odometry. You HAVE to use a good odometry estrimator as LOAM -->
    <node pkg="tf" type="static_transform_publisher" name="odom_tf" args="0 0 0 0 0 0 odom base_link_dll 10" />

    <param name="initialPoseByOdom_topic" value="/map_node/odometry" />
    <param name="in_cloud"                value="/lidar_odom_node/meta_cloud" />

    <param name="base_frame_id"   value="base_link_dll" />
    <param name="odom_frame_id"   value="odom" />
    <param name="global_frame_id" value="map" />

    <param name="use_imu" value="false" /> <!-- use_yaw_increments -->
      
    <param name="update_rate" value="10.0" />

    <!-- set Initial Pose -->
    <param name="initial_x"   value="0.0"/>
    <param name="initial_y"   value="0.0"/>
    <param name="initial_z"   value="0.0"/>
    <param name="initial_a"   value="0.0"/> <!-- yaw, attitude -->

    <param name="update_min_d"    value="0.01" />
    <param name="update_min_a"    value="0.01" /> 
    <param name="update_min_time" value="0.1" />

    <param name="align_method" value="1" />  <!-- 1: DLL, 2: NDT, 3: ICP -->

    <param name="map_path" value="$(find dll)/maps/guangzhou_port_map-0.5m-0.500000.bt" />
    <param name="publish_point_cloud" value="true" />    
    <!-- <param name="publish_grid_slice"  value="0.1" />  -->
    <!-- Launch DLL -->
    <node name="dll_node" type="dll_node" pkg="dll" output="screen">
      <!-- <remap from="/dll_node/initial_pose" to="/initialpose"/>
      <remap from="/imu" to="/airsim_node/base_link_dll/imu/Imu"/> -->
    </node>
  </group>

  <!-- <node name="rviz" type="rviz" pkg="rviz" args="-d $(find dll)/rviz/dll-guangzhou_port.rviz" output="screen"/> -->

  <!-- <node pkg="rosbag" type="play" name="rosbag_play" output="screen" args="/home/udi/SLAM_codes/dataset/guangzhou_port/localization-test-20250220-1-filtered.bag - -clock -l -d 10.0"/> -->

</launch>	
